synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -24,6 +24,8 @@
 from twisted.internet.task import LoopingCall
 from twisted.internet.threads import deferToThread
 
+from synapse.logging import context
```

Comment:
<sigh> This isn't really part of synapse's public API, though I think it's the best bet for now.

@babolivier did we decide what to do about this? Should we just make things like `run_as_background_process` part of the public API? Or should we add things like `ModuleApi.call_at_startup` to solve individual usecases?

Reply:
As an API consumer, I'd find the latter more agreeable. It's weird to use methods from two different libraries depending on whether I want to run the code asap or later.

Reply:
that's fair, though I'm also concerned that we'll never cover all usecases and people will *still* end up having to use `run_as_background_process` directly.

Reply:
Can't we do both? I can see usecases for both of them.